core/utils.h

```diff
@@ -1567,15 +1567,21 @@ enum { LONGJMP_EXCEPTION = 1 };
 
 #include "stats.h"
 
+#ifdef UNIX
+#    define UNUSED __attribute__((unused))
+#else
+#    define UNUSED
+#endif
+
 /* Use to shut up the compiler about an unused variable when the
  * alternative is a painful modification of more src code. Our
  * standard is to use this macro just after the variable is declared
  * and to use it judiciously.
  */
-#define UNUSED_VARIABLE(pv)           \
-    {                                 \
-        void *unused_pv = (void *)pv; \
-        unused_pv = NULL;             \
+#define UNUSED_VARIABLE(pv)                  \
```

Comment:
Why do we need UNUSED_VARIABLE and UNUSED both?

Edit: Oh okay. We use the former to not have to put too many pre-processor defines on the declaration. Whereas the latter is used when the variable is unused in all or only a small number of environments.

Reply:
s/only a small number of environments/simple set of environments/

Reply:
Also I think we didn't find a cross-platform compiler marking that worked well in the past: we were avoiding pragmas in VS